src/ops/cast.cc

```diff
@@ -63,26 +64,29 @@ Op *Cast::create_operator_from_layer(
 //   return hash;
 // }
 
+CastParams Cast::get_params() const {
+  CastParams params;
+  params.dtype = this->dtype;
+  return params;
+}
+
 using PCG::Node;
 Node FFModel::get_or_create_cast_node(const ParallelTensor input,
                                       DataType dtype) {
-  if (input->dims[input->num_dims - 1].degree != 1) {
-    return Node::INVALID_NODE;
-  }
-
-  size_t hash = input->get_owner_independent_hash();
-  hash_combine(hash, dtype);
+  CastParams params;
+  params.dtype = dtype;
+  return get_or_create_node<Cast>(input, params);
+}
 
-  Cast *cast;
-  auto const &it = this->cached_cast_ops.find(hash);
-  if (it != cached_cast_ops.end()) {
-    cast = it->second;
-  } else {
-    cast = new Cast(*this, input, dtype, NULL);
-    cached_cast_ops[hash] = cast;
-  }
+bool CastParams::is_valid(ParallelTensorShape const &input) const {
+  // TODO: more check on the input shape
```

Comment:
Are there more checks that should be done here? What are they?

Reply:
I forgot to delete this comment. See [here](https://github.com/flexflow/FlexFlow/blob/master/src/ops/cast.cc#L66) for reference. I think this is the only check required.

Reply:
Sounds good, then let's just make sure we remove the comment before we merge. On a side topic, do you have any idea of what the existing check is doing/why it's there?

Reply:
I'm not sure what the purpose of that check is. Actually, I don't think I could explain most of the checks that deal with `degree` and parallel dims. Before we merge this, it'd help to discuss each operator since some of the checks are insufficient as it is.